cumulus_library/studies/core/encounter.sql

```diff
@@ -56,7 +57,9 @@ SELECT DISTINCT
     p.ethnicity_display,
     p.postalcode3
 FROM temp_encounter AS e
-LEFT JOIN core__act_encounter_code_v3 AS ac ON ac.code = e.class.code
+LEFT JOIN core__fhir_mapping_expected_act_encounter_code_v3 AS eac
+    ON eac.found = e.class.code
+LEFT JOIN core__fhir_act_encounter_code_v3 AS ac ON eac.expected = ac.code
```

Comment:
Both these lines should also be checking the `system` yeah? Otherwise who knows what you are matching on.

Reply:
So, on the one hand, i agree; on the other, i think the namespace for this one is small enough that it likely doesn't matter.

I'd almost rather wait to see if someone responds to the request i made in slack for running a query from a different EHR and see how bad this looks in practice. But I can do it out of maximum parinoia.

Reply:
Oh OK let me pause approval until we iron this out - this is the only real concern I had.

What's the argument for not checking system? The `R` code could be used in some random EHR-vendor-specific-system and mean a totally different thing. You _have_ to always treat codes & systems as a cohesive unit. Otherwise the code doesn't mean anything.

Reply:
as an aside - i'm thinking about creating some kind of runnable artifact as part of this repo we can ask folks something like 'run this and send it back to us so we can confirm data issues' to gather information like this.

Reply:
We chatted out of band - here's my understanding of our convo, Matt, for posterity:

- We both want to eventually get to the place I described above.
- We can't do it today because some Epic sites provide the same _code_ but with an Epic-internal _system_ -- and the code in this PR allows for that. If we changed it now to require a system, we might be tossing some Epic data aside (though Mike thinking now -- we don't believe it's the existing Epic encounter system we use in library? `urn:oid:2.16.840.1.113883.4.642.3.248`)
- We're asking some sites to run some SQL to provide systems for their codes, and once that's in hand we can come back and update this.

So I'll re-approve. It sounds like we're both moving toward the same vision. I'm just more impatient than Matt.

Reply:
for posterity - we believe that that urn is :probably: equivalent to http://hl7.org/fhir/ValueSet/encounter-type, which is not what we're trying to use here.